File: user_struct.rs
// Define a User struct with name and age fields
#[derive(Debug, Clone)]
struct User {
    name: String,
    age: u32,
}

fn main() {
    // Create a sample user
    let user = User {
        name: String::from("Alice"),
        age: 30,
    };
    
    // Print the user for verification
    println!("Created user: {:?}", user);
    
    // Serialize and save to a file with a random name
    use std::fs::File;
    use std::io::Write;
    use rand::Rng;
    
    // Generate a random filename
    let mut rng = rand::thread_rng();
    let random_filename = format!("user_data_{}.txt", rng.gen::<u32>());
    
    // Write user data to the file
    let mut file = File::create(&random_filename).expect("Could not create file");
    writeln!(file, "Name: {}", user.name).expect("Could not write name");
    writeln!(file, "Age: {}", user.age).expect("Could not write age");
    
    println!("User data saved to: {}", random_filename);
}

// Add required dependencies
// Note: This example uses the rand crate for generating random filenames

File: user_struct.py
import json
import random
import string

# Define a User class with name and age fields (similar to Rust struct)
class User:
    def __init__(self, name, age):
        self.name = name
        self.age = age

    def to_dict(self):
        return {
            "name": self.name,
            "age": self.age
        }

def generate_random_filename(extension="json"):
    """Generate a random filename"""
    random_string = ''.join(random.choices(string.ascii_lowercase + string.digits, k=10))
    return f"user_data_{random_string}.{extension}"

def main():
    # Create a user instance
    user = User("Alice", 30)
    
    print(f"Created user: {user.name}, age {user.age}")
    
    # Convert user to dictionary and then to JSON
    user_data = user.to_dict()
    user_json = json.dumps(user_data, indent=2)
    
    # Generate a random filename
    random_filename = generate_random_filename()
    
    # Write user data to the file
    with open(random_filename, 'w') as f:
        f.write(user_json)
    
    print(f"User data saved to: {random_filename}")

if __name__ == "__main__":
    main()